src/util/path/path.ts

```diff
@@ -297,64 +333,64 @@ export const fromArcToBeziers = (
 };
 
 /**
- * This function take a parsed SVG path and make it simpler for fabricJS logic.
- * simplification consist of: only UPPERCASE absolute commands ( relative converted to absolute )
- * S converted in C, T converted in Q, A converted in C.
- * @param {PathData} path the array of commands of a parsed svg path for `Path`
- * @return {PathData} the simplified array of commands of a parsed svg path for `Path`
+ * This function takes a parsed SVG path and makes it simpler for fabricJS logic.
+ * Simplification consist of:
+ * - All commands converted to absolute (lowercase to uppercase)
+ * - S converted to C
+ * - T converted to Q
+ * - A converted to C
+ * @param {TComplexPathData} path the array of commands of a parsed SVG path for `Path`
+ * @return {TSimplePathData} the simplified array of commands of a parsed SVG path for `Path`
+ * TODO: figure out how to remove the type assertions in a nice way
  */
-export const makePathSimpler = (path: PathData): PathData => {
-  // x and y represent the last point of the path. the previous command point.
+export const makePathSimpler = (path: TComplexPathData): TSimplePathData => {
+  // x and y represent the last point of the path, AKA the previous command point.
   // we add them to each relative command to make it an absolute comment.
   // we also swap the v V h H with L, because are easier to transform.
   let x = 0,
     y = 0;
-  const len = path.length;
   // x1 and y1 represent the last point of the subpath. the subpath is started with
   // m or M command. When a z or Z command is drawn, x and y need to be resetted to
   // the last x1 and y1.
   let x1 = 0,
     y1 = 0;
   // previous will host the letter of the previous command, to handle S and T.
   // controlX and controlY will host the previous reflected control point
-  let destinationPath: PathData = [],
-    previous,
-    controlX,
-    controlY;
-  for (let i = 0; i < len; ++i) {
-    let converted = false;
-    const current = path[i].slice(0);
+  const destinationPath: TSimplePathData = [];
+  let previous: string | undefined,
+    // placeholders
+    controlX = 0,
+    controlY = 0;
+  for (const parsedCommand of path) {
+    const current: TComplexParsedCommand = Object.assign([], parsedCommand);
```

Comment:
`[...parsedCommand]` vs `Object.assign([], parsedCommand)`. slice(0) or concat() are good too.

Reply:
Oh I forgot about `[...parsed]`, I like that a lot more and I'll make that change really quick